sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/converter/RubyVisitor.java

```diff
@@ -120,6 +128,61 @@ public IRubyObject onDefs(ThreadContext context, IRubyObject node) {
     return createFunctionDeclaration(context, node);
   }
 
+  @JRubyMethod(name = "on_if")
+  public IRubyObject onIf(ThreadContext context, IRubyObject node) {
+    IRubyObject updatedNode = visitNode(context, "on_if", node);
+
+    Token mainKeyword = noLocation(updatedNode, "keyword") ? null : getMetaData(updatedNode, "keyword").tokens().get(0);
+    if (mainKeyword == null || mainKeyword.text().equals("unless")) {
+      // Ternary operator and "unless" are not considered as "IfTree" for now
+      return createNativeTreeAdapter(updatedNode);
```

Comment:
IMO it's cleaner to return `node` (then `process` should do the trick of native creation)